odoo/sql_db.py

```diff
@@ -22,6 +22,7 @@
 import psycopg2.extensions
 from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT, ISOLATION_LEVEL_READ_COMMITTED, ISOLATION_LEVEL_REPEATABLE_READ
 from psycopg2.pool import PoolError
+from psycopg2.sql import SQL, Identifier
```

Comment:
IIRC `Identifier` takes whatever string in input and sanitize it so it can be used as a table name or a column name am I right ? What does `SQL` do ?

Reply:
It allows you to build [safe, composable SQL queries](https://www.psycopg.org/docs/sql.html), iirc you cannot pass an `Identifier` object directly to `cr.execute`, you must first build an `SQL` object and pass that to `cr.execute`

Reply:
> IIRC Identifier takes whatever string in input and sanitize it so it can be used as a table name or a column name am I right ?

Well it really considers the data passed in to be an SQL identifier so that it can escape / quote / ... it correctly as necessary when composed into an SQL.

> What does `SQL` do ?

It actually does the composition, `SQL` marks the datum as SQL query snippets, which either get formatted into other query snippets, or have other composables (identifiers, literals, or placeholders) composed in.